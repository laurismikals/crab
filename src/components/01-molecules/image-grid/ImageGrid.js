import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import map from 'lodash/fp/map';

import LoadingSpinner from '../../00-atoms/loading-spinner/LoadingSpinner';
import Image from '../../00-atoms/image/Image';

import './ImageGrid.css';

export class ImageGrid extends PureComponent {
  static propTypes = {
    images: PropTypes.arrayOf(PropTypes.shape),
    fetching: PropTypes.bool,
  };
  static defaultProps = {
    images: [],
    fetching: false,
  };
  render() {
    const {
      fetching,
      images,
    } = this.props;

    return (
      <div
        className="image-grid"
        style={fetching ? {minHeight: 50} : {}}
      >
        {fetching && <LoadingSpinner/>}
        {images &&
        <div className="width-100 margin-bottom-30">
          <div className="row row--gutters-30">
            {map(({id, image, width, height}) => (
              <div key={id} className="col-s1-24 col-s2-12 col-s4-8 col-s7-6 flex justify--center flex-align--center">
                <Image
                  className="margin-bottom-30"
                  src={image}
                  alt=""
                  x={width}
                  y={height}
                />
              </div>
            ))(images)}
          </div>
        </div>
        }
      </div>
    )
  }
}

export default ImageGrid;
