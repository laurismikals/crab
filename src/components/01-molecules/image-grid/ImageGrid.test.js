import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import { ImageGrid } from './ImageGrid';

describe('Organisms: ImageGrid', () => {
  const minProps = {
    images: [
      {
        id: 1,
        image: 'image.webp',
        width: 100,
        height: 100,
      },
      {
        id: 2,
        image: 'image1.webp',
        width: 100,
        height: 100,
      }
    ],
    fetching: false,
  };

  it('renders correctly', () => {
    const tree = shallow(
      <ImageGrid />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when fetching', () => {
    const tree = shallow(
      <ImageGrid fetching={true} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with data', () => {
    const tree = shallow(
      <ImageGrid {...minProps} />
    );
    expect(tree).toMatchSnapshot();
  });
});
