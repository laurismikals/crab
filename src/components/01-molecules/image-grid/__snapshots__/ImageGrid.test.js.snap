// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: ImageGrid renders correctly 1`] = `
<div
  className="image-grid"
  style={Object {}}
>
  <div
    className="width-100 margin-bottom-30"
  >
    <div
      className="row row--gutters-30"
    />
  </div>
</div>
`;

exports[`Organisms: ImageGrid renders correctly when fetching 1`] = `
<div
  className="image-grid"
  style={
    Object {
      "minHeight": 50,
    }
  }
>
  <LoadingSpinner />
  <div
    className="width-100 margin-bottom-30"
  >
    <div
      className="row row--gutters-30"
    />
  </div>
</div>
`;

exports[`Organisms: ImageGrid renders correctly with data 1`] = `
<div
  className="image-grid"
  style={Object {}}
>
  <div
    className="width-100 margin-bottom-30"
  >
    <div
      className="row row--gutters-30"
    >
      <div
        className="col-s1-24 col-s2-12 col-s4-8 col-s7-6 flex justify--center flex-align--center"
        key="1"
      >
        <Image
          alt=""
          className="margin-bottom-30"
          src="image.webp"
          x={100}
          y={100}
        />
      </div>
      <div
        className="col-s1-24 col-s2-12 col-s4-8 col-s7-6 flex justify--center flex-align--center"
        key="2"
      >
        <Image
          alt=""
          className="margin-bottom-30"
          src="image1.webp"
          x={100}
          y={100}
        />
      </div>
    </div>
  </div>
</div>
`;
